buildSrc/src/main/kotlin/ServiceLoader.kt

```diff
@@ -0,0 +1,115 @@
+import org.gradle.api.Project
+import software.amazon.smithy.aws.traits.ServiceTrait
+import software.amazon.smithy.model.Model
+import software.amazon.smithy.model.shapes.ServiceShape
+import software.amazon.smithy.model.traits.TitleTrait
+import java.io.File
+import kotlin.streams.toList
+
+/**
+ * Discovers services from the `aws-models` directory within the project.
+ *
+ * Since this function parses all models, it is relatively expensive to call. The result should be cached in a property
+ * during build.
+ */
+fun Project.discoverServices(serviceMembership: Membership): List<AwsService> {
+    val models = project.file("aws-models")
+    val baseServices = fileTree(models)
+        .sortedBy { file -> file.name }
+        .mapNotNull { file ->
+            val model = Model.assembler().addImport(file.absolutePath).assemble().result.get()
+            val services: List<ServiceShape> = model.shapes(ServiceShape::class.java).sorted().toList()
+            if (services.size > 1) {
+                throw Exception("There must be exactly one service in each aws model file")
+            }
+            if (services.isEmpty()) {
+                logger.info("${file.name} has no services")
+                null
+            } else {
+                val service = services[0]
+                val title = service.expectTrait(TitleTrait::class.java).value
+                val sdkId = service.expectTrait(ServiceTrait::class.java).sdkId
+                    .toLowerCase()
+                    .replace(" ", "")
+                    // TODO: the smithy models should not include the suffix "service"
```

Comment:
Is this TODO comment still relevant?

Reply:
yeah as of the current aws-models the service suffix is still included